### ISSUE TITLE
Improvements to symbolic module!

### ISSUE THREAD

**Issue by ritzvik** [einsteinpy/einsteinpy#405]
🐞 **Problem**
Lots of features are currently not available in symbolic module. I'm listing them and would continue to expand this list as things strike me! When working on a specific addition, kindly open an issue referencing to this one, and then open a PR. Mainly, the modifications would be required in `BaseRelativityTensor` class.

🎯 **Goal**

  - [ ] Wedge Product
  - [ ] Exterior Derivative
  - [ ] Covariant Derivative
  - [ ] Derivative (although, plain derivative is not a tensor, just a (n+1) order array)
  - [ ] Native function for Tensor Product

**Comment by abhijeetmanhas** [comment]
For covariant derivative, is it we have to consider levi-civita connection of reimannian geometry?

**Comment by kauii8school** [comment]
This may be relavent: https://docs.sympy.org/latest/modules/diffgeom.html. Is it ok to just use Sympy methods or should we implement native methods for all of these?

**Comment by ritzvik** [comment]
Yes ! This seems relevant. We would use the functions as much as possible. 
I didn’t know about this.

Regards,
Ritwik

**Comment by Starninja666** [comment]
Can I work on this?

**Comment by ritzvik** [comment]
> I have started working on this..!

This is a huge tranche of issues. Could you give some specificity about the issue you are solving?

**Comment by bibek22** [comment]
Hey @ritzvik, should I reference this issue itself or create a new issue for each goals listed above ?

**Comment by JeS24** [comment]
This may be a good reference: http://gravity.psu.edu/local/internal/blackholes_2012.pdf (Check the appendix and first few chapters).
